app/views/RoomInfoEditView/index.tsx

```diff
@@ -1,268 +1,131 @@
 import { MultiSelect } from '../../containers/UIKit/MultiSelect';
-import {
-	IApplicationState,
-	IBaseScreen,
-	IRoomSettings,
-	ISubscription,
-	SubscriptionType,
-	TSubscriptionModel
-} from '../../definitions';
-import { ERoomType } from '../../definitions/ERoomType';
+import { IBaseScreen, IRoomSettings, ISubscription, SubscriptionType } from '../../definitions';
 import I18n from '../../i18n';
-import database from '../../lib/database';
 import KeyboardView from '../../containers/KeyboardView';
-import { TSupportedPermissions } from '../../reducers/permissions';
 import { ModalStackParamList } from '../../stacks/MasterDetailStack/types';
 import { ChatsStackParamList } from '../../stacks/types';
-import { withTheme } from '../../theme';
+import { useTheme } from '../../theme';
 import EventEmitter from '../../lib/methods/helpers/events';
 import log, { events, logEvent } from '../../lib/methods/helpers/log';
 import { MessageTypeValues } from './messageTypes';
 import scrollPersistTaps from '../../lib/methods/helpers/scrollPersistTaps';
 import sharedStyles from '../Styles';
 import styles from './styles';
 import SwitchContainer from './SwitchContainer';
-import {
-	getRoomTitle,
-	hasPermission,
-	compareServerVersion,
-	showConfirmationAlert,
-	showErrorAlert,
-	random,
-	isAndroid
-} from '../../lib/methods/helpers';
+import { getRoomTitle, compareServerVersion, showErrorAlert, isAndroid, random } from '../../lib/methods/helpers';
 import { Services } from '../../lib/services';
 import Button from '../../containers/Button';
+import useRoomSubscription from './hooks/useRoomSubscription';
+import useRoomDeletionActions from './hooks/useRoomDeletionActions';
 
-interface IRoomInfoEditViewState {
-	room: ISubscription;
-	permissions: { [key in TSupportedPermissions]?: boolean };
-	name: string;
-	description?: string;
-	topic?: string;
-	announcement?: string;
-	joinCode: string;
-	nameError: any;
-	t: boolean;
-	ro: boolean;
-	reactWhenReadOnly?: boolean;
-	archived: boolean;
-	systemMessages?: boolean | string[];
-	enableSysMes?: boolean | string[];
-	encrypted?: boolean;
-}
-
-interface IRoomInfoEditViewProps extends IBaseScreen<ChatsStackParamList | ModalStackParamList, 'RoomInfoEditView'> {
-	serverVersion?: string;
-	encryptionEnabled: boolean;
-	setReadOnlyPermission: string[];
-	setReactWhenReadOnlyPermission: string[];
-	archiveRoomPermission: string[];
-	unarchiveRoomPermission: string[];
-	deleteCPermission: string[];
-	deletePPermission: string[];
-	deleteTeamPermission: string[];
-}
+interface IRoomInfoEditViewProps extends IBaseScreen<ChatsStackParamList | ModalStackParamList, 'RoomInfoEditView'> {}
 
 const MESSAGE_TYPE_VALUES = MessageTypeValues.map(m => ({
 	value: m.value,
 	text: { text: I18n.t('Hide_type_messages', { type: I18n.t(m.text) }) }
 }));
-
-class RoomInfoEditView extends React.Component<IRoomInfoEditViewProps, IRoomInfoEditViewState> {
-	randomValue = random(15);
```

Comment:
This used to be called on root level of a class component, similar to a `useEffect` without deps.